packages/backend-defaults/src/entrypoints/rootHttpRouter/rootHttpRouterServiceFactory.ts

```diff
@@ -55,72 +58,106 @@ export interface RootHttpRouterConfigureContext {
  *
  * @public
  */
-export type RootHttpRouterFactoryOptions = {
+export type StartRootHttpServerOptions = {
   /**
    * The path to forward all unmatched requests to. Defaults to '/api/app' if
    * not given. Disables index path behavior if false is given.
    */
   indexPath?: string | false;
 
   configure?(context: RootHttpRouterConfigureContext): void;
+
+  config: RootConfigService;
+  logger: RootLoggerService;
+  lifecycle: RootLifecycleService;
+  health: RootHealthService;
 };
 
-function defaultConfigure({ applyDefaults }: RootHttpRouterConfigureContext) {
-  applyDefaults();
+/**
+ * @deprecated Use {@link StartRootHttpServerOptions} instead.
+ * @public
+ */
+export type RootHttpRouterFactoryOptions = StartRootHttpServerOptions;
+
+/**
+ * Creates and starts a root HTTP server with the given options.
+ *
+ * @public
+ */
+export async function startRootHttpServer(options: StartRootHttpServerOptions) {
+  const {
+    indexPath,
+    configure = (ctx: RootHttpRouterConfigureContext) => ctx.applyDefaults(),
+    config,
+    lifecycle,
+    health,
+  } = options ?? {};
+
+  const logger = options.logger.child({ service: 'rootHttpRouter' });
+  const app = express();
+
+  const router = DefaultRootHttpRouter.create({ indexPath });
+  const middleware = MiddlewareFactory.create({ config, logger });
+  const routes = router.handler();
+
+  const healthRouter = createHealthRouter({ health });
+  const server = await createHttpServer(
+    app,
+    readHttpServerOptions(config.getOptionalConfig('backend')),
+    { logger },
+  );
+
+  configure({
+    app,
+    server,
+    routes,
+    middleware,
+    config,
+    logger,
+    lifecycle,
+    healthRouter,
+    applyDefaults() {
+      app.use(middleware.helmet());
+      app.use(middleware.cors());
+      app.use(middleware.compression());
+      app.use(middleware.logging());
+      app.use(healthRouter);
+      app.use(routes);
+      app.use(middleware.notFound());
+      app.use(middleware.error());
+    },
+  });
+
+  lifecycle.addShutdownHook(() => server.stop());
+
+  await server.start();
+
+  return router;
+}
+
+/**
+ * @public
+ */
+export namespace startRootHttpServer {
+  /**
+   * Dependency map for `startRootHttpServer`, to be passed to the `deps` options of `createServiceFactory`.
+   * @public
+   */
+  export const deps = {
+    config: coreServices.rootConfig,
+    logger: coreServices.rootLogger,
+    lifecycle: coreServices.rootLifecycle,
+    health: coreServices.rootHealth,
+  };
 }
```

Comment:
Here's the dependency map bit, happy to roll this back and just ask users to list explicitly as well.